smart-contracts-tutorial/guess37-multiple-participants.md

```diff
@@ -0,0 +1,114 @@
+# Guess37 multiple participants
+
+In the [previous step](./first-contract), we set up a contract and run it and played the guess37 game
+
+In this section, we'll see how to play 2 different roles: "contract installer" and "game player"
+
+
+## Setting up 2 contract participants on your machine
+
+In real life, different participants will participate to the contract from different machines located potentially anywhere on the planet. For the purpose of learning how to design, write and test contracts, it is easier to set up 2 participants to the contract on your machine
+
+To setup 2 participants, you need to [build for cosmic-swingset scenario 2](https://github.com/Agoric/cosmic-swingset#build-from-source) and run scenario 2 this way:
+```sh
+make scenario2-setup NUM_SOLOS=2
+make scenario2-run-chain
+```
+
+Wait about 5 seconds for the chain to produce its first block, then switch to another terminal:
+
+`make scenario2-run-client BASE_PORT=8000`\
+and then in a third terminal:\
+`make scenario2-run-client BASE_PORT=8001`
+
+The first participant (contract installer) interacts with the smart contract on the blockchain via http://localhost:8000/
+The second participant (game player) interacts with the smart contract on the blockchain via http://localhost:8001/ (notice the port difference)
+
+
+## Acting as the contract installer
+
+Open http://localhost:8000/ in a browser tab and in the REPL enter the following commands in order:
+```
+home
+host = home~.contractHost
+installation = host~.install({start: home.guess37ContractSource})
+invites = installation~.spawn()
+playerInvite = invites~.playerInvite
+```
+
+The contract is now setup, like it was in the previous step. Let's now share the player invite with the game player:
+```
+handoff = home~.handoff
+board = handoff~.createBoard("guess37 game")
+board~.addEntry("invite", playerInvite)
+```
+
+Then, the contract installer would communicate to the game player that a board called `guess37 game` with an entry called `invite` is available for them to retrieve the invite
+This communication would happen out-of-band. Maybe via email, Mastodon DM, SMS, hand-written note; whatever is best suited to share these secrets
+
+`handoff` is a convenience [provided by ERTP](https://github.com/Agoric/ERTP/blob/master/more/handoff/handoff.js) and exposed publicly so that people can securely share secrets and capabilities via the blockchain
+
+
+## Acting as the game player
+
+Open http://localhost:8001/ in a different browser tab and in the REPL enter the following commands in order:
+```
+home
+handoff = home~.handoff
+board = handoff~.grabBoard("guess37 game")
+playerInvite = board~.lookup("invite")
+```
+
+Now we have the invite to participate to the contract. Let's play the game:
+```
+seat = host~.redeem(playerInvite)
```

Comment:
`host` is undefined

Reply:
did this show up as undefined on line 33?

Reply:
no but line 33 is the contract installer role and line 64 is the game player role. They are 2 different solo nodes, accessed through 2 different tabs (different ports), so are 2 different contexts
I would imagine REPL variables are independent (i haven't verified). At least, i will document as if they were :see_no_evil: